src/main/java/org/opensearch/security/action/tenancy/DefaultTenantUpdateTransportAction.java

```diff
@@ -0,0 +1,106 @@
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.action.tenancy;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.client.Client;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.security.configuration.ConfigurationRepository;
+import org.opensearch.security.dlic.rest.api.AbstractApiAction;
+import org.opensearch.security.securityconf.impl.CType;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+import org.opensearch.security.securityconf.impl.v7.ConfigV7;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.tasks.Task;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+
+public class DefaultTenantUpdateTransportAction extends HandledTransportAction<StringSettingUpdateRequest, StringSettingRetrieveResponse> {
+
+    private static final Logger log = LogManager.getLogger(DefaultTenantUpdateTransportAction.class);
+
+    private final String securityIndex;
+    private final ConfigurationRepository config;
+    private final Client client;
+    private final ThreadPool pool;
+
+    @Inject
+    public DefaultTenantUpdateTransportAction(final Settings settings,
+                                             final TransportService transportService,
+                                             final ActionFilters actionFilters,
+                                             final ConfigurationRepository config,
+                                             final ThreadPool pool,
+                                             final Client client) {
+        super(DefaultTenantUpdateAction.NAME, transportService, actionFilters, StringSettingUpdateRequest::new);
+
+        this.securityIndex = settings.get(ConfigConstants.SECURITY_CONFIG_INDEX_NAME, ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+
+        this.config = config;
+        this.client = client;
+        this.pool = pool;
+    }
+
+    /** Load the configuration from the security index and return a copy */
+    protected final SecurityDynamicConfiguration<?> load() {
+        return config.getConfigurationsFromIndex(Collections.singleton(CType.CONFIG), false).get(CType.CONFIG).deepClone();
+    }
+
+
+
+    @Override
+    protected void doExecute(final Task task, final StringSettingUpdateRequest request, final ActionListener<StringSettingRetrieveResponse> listener) {
+
```

Comment:
All of the transport action update methods have nearly the same code, lets refactor so its only written once.  I think you could try something like the following.

```java
public ??? doUpdate(Consumer<ConfigV7> updateAction, ...)`
```

Then for this method it could look something like...

```java
@Override
protected void doExecute(final Task task, final StringSettingUpdateRequest request, final ActionListener<StringSettingRetrieveResponse> listener) {
    doUpdate(
        (config) -> { config.dynamic.kibana.default_tenant = request.getValue(); },
        DefaultTenantRetrieveAction.INSTANCE,
        listener
    );
}
```

Reply:
@peternied As per suggestions from @anijain-Amazon, I am implementing one single API and therefore single class for updating all properties.